src/commands/general/birthday.ts

```diff
@@ -0,0 +1,158 @@
+import { CommandDefinition } from '../../lib/command';
+import { Roles, CommandCategory } from '../../constants';
+import { makeEmbed } from '../../lib/embed';
+import { getConn } from '../../lib/db';
+import { Channels } from '../../constants';
+
+const permittedRoles = [
+    Roles.ADMIN_TEAM,
+    Roles.MODERATION_TEAM,
+    Roles.DEVELOPMENT_TEAM,
+    Roles.MEDIA_TEAM,
+    Roles.FBW_EMERITUS,
+];
+
+export const birthday: CommandDefinition = {
+    name: 'birthday',
+    description: 'Manages birthday reminders',
+    category: CommandCategory.GENERAL,
+    executor: async (msg) => {
+        // Allow commands only in birthday thread
+        if(msg.channel.id != Channels.BIRTHDAY_THREAD)
+        {
+            return;
+        }
+
+        const conn = await getConn();
+        const args: string[] = msg.content.split(' ').slice(1);
+
+        let birthdayEmbed;
+
+        const hasPermittedRole = msg.member.roles.cache.some((role) => permittedRoles.map(r => r.toString()).includes(role.id));
+
+        if (!hasPermittedRole) {
+            birthdayEmbed = makeEmbed({
+                title: 'Birthday reminder',
+                description: 'You do not have permission to use this command.',
+                color: 'RED',
+            });
+        } else {
+            if (args[0] === 'add' || args[0] === 'set') {
+                const user = msg.mentions.users.first();
+
+                if (!user) {
+                    birthdayEmbed = makeEmbed({
+                        title: 'Birthday reminder',
+                        description: 'You need to mention a user to add a birthday reminder',
+                        color: 'RED',
+                    });
+                } else {
+                    let birthdayDay;
+                    let birthdayMonth;
+
+                    let birthdayStrings = args.slice(2).join('/').split(/[\/-]/);
+
+                    // Catch invalid dates
+                    if (birthdayStrings.length < 2) {
+                        birthdayEmbed = makeEmbed({
+                            title: 'Birthday reminder',
+                            description: 'Invalid date format. Please use `.birthday add <user> <month>/<day>`',
```

Comment:
Would appropriate error handling not be to say not enough args were provided instead of incorrect date format?

Reply:
That definitely works too, since the args in question are just month and day